lib/unexpected-snapshot-update.js

```diff
@@ -0,0 +1,176 @@
+        columnNumber: parseInt(matchTopFrame[3], 10)
+      };
+    }
+  }
+}
+
+module.exports = {
+  name: 'unexpected-snapshot',
+  version: require('../package.json').version,
+  installInto(expect) {
+    const expectForRendering = expect.child();
+
+    if (typeof Buffer === 'function') {
+      expectForRendering.addType({
+        name: 'infiniteBuffer',
+        base: 'Buffer',
+        identify(obj) {
+          return this.baseType.identify(obj);
+        },
+        inspect(value, depth, output, inspect) {
+          if (value.length > 32) {
+            return output.code(
+              `Buffer.from('${value.toString('base64')}', 'base64')`,
+              'javascript'
+            );
+          } else {
+            return this.baseType.inspect.call(
+              this,
+              value,
+              depth,
+              output,
+              inspect
+            );
+          }
+        },
+        prefix(output) {
+          return output.code('Buffer.from([', 'javascript');
+        },
+        suffix(output) {
+          return output.code('])', 'javascript');
+        },
+        hexDumpWidth: Infinity // Prevents Buffer instances > 16 bytes from being truncated
+      });
+    }
+
+    expectForRendering.addType({
+      base: 'Error',
+      name: 'overriddenError',
+      identify(obj) {
+        return this.baseType.identify(obj);
+      },
+      inspect(value, depth, output, inspect) {
+        var obj = _.extend({}, value);
+
+        var keys = Object.keys(obj);
+        if (keys.length === 0) {
+          output
+            .text('new Error(')
+            .append(inspect(value.message || ''))
+            .text(')');
+        } else {
+          output
+            .text('(function () {')
+            .text(`var err = new ${value.constructor.name || 'Error'}(`)
+            .append(inspect(value.message || ''))
+            .text(');');
+          keys.forEach(function(key, i) {
+            output.sp();
+            if (/^[a-z$_][a-z0-9$_]*$/i.test(key)) {
+              output.text(`err.${key}`);
+            } else {
+              output
+                .text('err[')
+                .append(inspect(key))
+                .text(']');
+            }
+            output
+              .text(' = ')
+              .append(inspect(obj[key]))
+              .text(';');
+          });
+          output.sp().text('return err;}())');
+        }
+      }
+    });
+
+    const symbol = Symbol('unexpectedSnapshot');
+
+    const topLevelFixes = [];
+
+    expect.addAssertion(
+      ['<any> to equal snapshot', '<any> to inspect as snapshot'],
+      (expect, subject, ...args) => {
+        if (expect.context[symbol]) {
+          topLevelFixes.push({
+            ...expect.context[symbol],
+            subject,
+            assertionName: expect.testDescription,
+            status: 'missing'
+          });
+          require('./ensureAfterHookIsRegistered')(topLevelFixes);
```

Comment:
This require doesn't have to be "lazy" now that the containing file is not in the browser build anymore :tada:

You can just move it to the top.

We could also consider merging `ensureAfterHookIsRegistered.js` into `unexpected-snapshot-update.js`. The separation is a bit arbitrary, and IIRC I did it in an attempt to make the browser build work :)